core/domain/event_services.py

```diff
@@ -75,35 +76,21 @@ def _notify_continuous_computation_listeners_async(cls, *args, **kwargs):
 
     @classmethod
     def _handle_event(cls, exploration_id, exploration_version, state_name,
-            rule_spec_string, session_id, time_spent_in_secs, params, answer):
-        """Records an event when an answer triggers a rule."""
+            answer_group_index, rule_spec_index, classification_categorization,
+            session_id, time_spent_in_secs, params, normalized_answer):
+        """Records an event when an answer triggers a rule. The answer recorded
+        here is a Python-representation of the actual answer submitted by the
+        user.
+        """
         # TODO(sll): Escape these args?
-        # TODO(msl): remove old answer recording models.
-
-        # In the old framework, answers are converted to unicode even when they
-        # are not strings. This will be removed when migrating to the new
-        # framework.
-        stats_models.process_submitted_answer(
-            exploration_id, exploration_version, state_name,
-            rule_spec_string, unicode(answer))
+        exploration = exp_services.get_exploration_by_id(
+            exploration_id, version=exploration_version)
+        interaction_id = exploration.states[state_name].interaction.id
         stats_services.record_answer(
-            exploration_id, exploration_version, state_name, rule_spec_string,
-            session_id, time_spent_in_secs, params, answer)
-
-
-class DefaultRuleAnswerResolutionEventHandler(BaseEventHandler):
-    """Event handler for recording resolving of answers triggering the default
-    rule."""
-
-    EVENT_TYPE = feconf.EVENT_TYPE_DEFAULT_ANSWER_RESOLVED
-
-    @classmethod
-    def _handle_event(cls, exploration_id, state_name, answers):
-        """Resolves a list of answers for the default rule of this state."""
-        # TODO(sll): Escape these args?
-        stats_models.resolve_answers(
-            exploration_id, state_name,
-            exp_domain.DEFAULT_RULESPEC_STR, answers)
+            exploration, state_name, stats_domain.SubmittedAnswer(
```

Comment:
Mm, not great to pass the big exploration blob all over the place. Can't you just pass the necessary individual pieces of data?